pydantic/json_schema.py

```diff
@@ -1043,17 +1043,17 @@ def default_schema(self, schema: core_schema.WithDefaultSchema) -> JsonSchemaVal
         #     return json_schema
 
         # we reflect the application of custom plain, no-info serializers to defaults for
-        # json schemas viewed in serialization mode
+        # JSON Schemas viewed in serialization mode:
         # TODO: improvements along with https://github.com/pydantic/pydantic/issues/8208
-        # TODO: improve type safety here
-        if self.mode == 'serialization':
-            if (
-                (ser_schema := schema['schema'].get('serialization', {}))
-                and (ser_func := ser_schema.get('function'))
-                and ser_schema.get('type') == 'function-plain'  # type: ignore
-                and ser_schema.get('info_arg') is False  # type: ignore
-            ):
-                default = ser_func(default)  # type: ignore
+        if (
+            self.mode == 'serialization'
+            and (ser_schema := schema['schema'].get('serialization'))
+            and (ser_func := ser_schema.get('function'))
+            and ser_schema.get('type') == 'function-plain'
+            and not ser_schema.get('info_arg')
```

Comment:
Switching for `and ser_schema.get('info_arg') is False` releaved an issue in the tests, that I will fix in this PR.